src/Universum/List.hs

```diff
@@ -1,11 +1,20 @@
-{-# LANGUAGE Safe #-}
+{-# LANGUAGE NoImplicitPrelude, ScopedTypeVariables #-}
 
 -- | Utility functions to work with lists.
 
 module Universum.List
        ( module Universum.List.Reexport
        , module Universum.List.Safe
+       , foldl'
        ) where
 
 import Universum.List.Reexport
 import Universum.List.Safe
+
+import GHC.Base (foldr, oneShot, seq, id)
+
+-- Changes the argument order of GHC.List's implementation of foldl' 
+foldl' :: forall a b . (a -> b -> b) -> b -> [a] -> b
+{-# INLINE foldl' #-}
+foldl' k z0 xs = 
+    foldr (\(v::a) (fn::b->b) -> oneShot (\(z::b) -> z `seq` fn (k v z))) (id :: b -> b) xs z0
```

Comment:
Ideally this function should replace `foldl'` in `Container` type class because we want it to work for every type, not only list. And the idea was not to copy-paste implementation from `base` package but just have

```haskell
foldl' f = GHC.Base.foldl' (flip f)
{-# INLINE foldl' #-}
```

and see how well GHC doing his job in optimization and inlining.